File: code-example-tests/csharp/driver/Tests/Aggregation/Pipelines/TutorialTests.cs
using DotNetEnv;
using MongoDB.Driver;
using Utilities.Comparison;

namespace Tests.Aggregation.Pipelines;

public class TutorialTests
{
    private IMongoClient _client;
    private string examplesFolder;

    [SetUp]
    [Description("Initializes the MongoDB client before each test")]
    public void Setup()
    {
        var connectionString = Env.GetString("CONNECTION_STRING",
            "Env variable not found. Verify you have a .env file with a valid connection string.");
        _client = new MongoClient(connectionString);
        examplesFolder = $"{Directory.GetCurrentDirectory()}/../../../../Examples/Aggregation/OutputFiles/";
    }

    [Test]
    [Description("Tests that Aggregation Filter pipeline output matches the expected documentation examples with ordered sorting")]
    public void TestFilterOutputMatchesDocs()
    {
        var example = new Examples.Aggregation.Pipelines.Filter.Tutorial();
        example.LoadSampleData();
        var results = example.PerformAggregation();
        var fullPath = examplesFolder + "FilterTutorialOutput.txt";

        Expect.That(fullPath)
            .WithOrderedSort()
            .ShouldMatch(results);
    }

    [Test]
    [Description("Tests that Aggregation Group pipeline output matches the expected documentation examples with ordered sorting")]
    public void TestGroupOutputMatchesDocs()
    {
        var example = new Examples.Aggregation.Pipelines.Group.Tutorial();
        example.LoadSampleData();
        var results = example.PerformAggregation();

        var fullPath = examplesFolder + "GroupTutorialOutput.txt";

        Expect.That(fullPath)
            .WithOrderedSort()
            .ShouldMatch(results);

    }

    [Test]
    [Description("Tests that Aggregation Unwind pipeline output matches the expected documentation examples")]
    public void TestUnwindOutputMatchesDocs()
    {
        var example = new Examples.Aggregation.Pipelines.Unwind.Tutorial();
        example.LoadSampleData();
        var results = example.PerformAggregation();

        var fullPath = examplesFolder + "UnwindTutorialOutput.txt";

        Expect.That(fullPath).ShouldMatch(results);
    }

    [Test]
    [Description("Tests that Aggregation Join one-to-one pipeline output matches the expected documentation examples with ordered sorting")]
    public void TestJoinOneToOneOutputMatchesDocs()
    {
        var example = new Examples.Aggregation.Pipelines.JoinOneToOne.Tutorial();
        example.LoadSampleData();
        var results = example.PerformAggregation();

        var fullPath = examplesFolder + "JoinTutorialOutput.txt";

        Expect.That(fullPath)
            .WithOrderedSort()
            .ShouldMatch(results);
    }

    [Test]
    [Description("Tests that Aggregation Join multi-field pipeline output matches the expected documentation examples with ordered sorting")]
    public void TestJoinMultiFieldOutputMatchesDocs()
    {
        var example = new Examples.Aggregation.Pipelines.JoinMultiField.Tutorial();
        example.LoadSampleData();
        var results = example.PerformAggregation();

        var fullPath = examplesFolder + "JoinMultifieldTutorialOutput.txt";

        Expect.That(fullPath)
            .WithOrderedSort()
            .ShouldMatch(results);
    }

    [TearDown]
    [Description("Cleans up the test database and disposes the MongoDB client after each test")]
    public void TearDown()
    {
        // Drop the database after the test completes
        _client.DropDatabase("agg_tutorials_db");
        _client.Dispose();
    }
}

File: code-example-tests/java/driver-sync/src/test/java/aggregation/pipelines/TutorialTests.java
package aggregation.pipelines;

import com.mongodb.client.*;
import org.bson.Document;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import mongodb.comparison.Expect;

import java.nio.file.Path;
import java.util.List;

public class TutorialTests {
    private final String uri = System.getenv("CONNECTION_STRING");
    private MongoClient mongoClient;
    private MongoDatabase database;

    @BeforeEach
    void setUp() {
        mongoClient = MongoClients.create(uri);
        database = mongoClient.getDatabase("agg_tutorials_db");
    }

    @AfterEach
    void tearDown() {
        database.drop();
        mongoClient.close();
    }

    @Test
    @DisplayName("Test that agg tutorial template app code runs and returns a specific document")
    void TestAggTutorial() {
        var output = AggTutorial.runExample(new String[]{"example arg"});
        var outputString = output.get(0).get("stringValue");
        var expectedOutput = "sample2";
        Expect.that(outputString)
            .shouldMatch(expectedOutput);
    }

    @Test
    @DisplayName("Test that filter aggregation pipeline matches output")
    void TestFilter() {
        var example = new aggregation.pipelines.filter.Tutorial();
        example.loadSampleData();
        var output = example.runTutorial();

        Expect.that(output)
            .shouldMatch("src/main/java/aggregation/pipelines/filter/TutorialOutput.txt");
    }

    @Test
    @DisplayName("Test that group aggregation pipeline matches output")
    void TestGroup() {
        var example = new aggregation.pipelines.group.Tutorial();
        example.loadSampleData();
        var output = example.runTutorial();

        Expect.that(output)
            .shouldMatch("src/main/java/aggregation/pipelines/group/TutorialOutput.txt");
    }

    @Test
    @DisplayName("Test that unwind aggregation pipeline matches output")
    void TestUnwind() {
        var example = new aggregation.pipelines.unwind.Tutorial();
        example.loadSampleData();
        var output = example.runTutorial();

        Expect.that(output)
            .shouldMatch("src/main/java/aggregation/pipelines/unwind/TutorialOutput.txt");
    }

    @Test
    @DisplayName("Test that one-to-one join aggregation pipeline matches output")
    void TestJoinOneToOne() {
        var example = new aggregation.pipelines.join1to1.Tutorial();
        example.loadSampleData();
        var output = example.runTutorial();

        Expect.that(output)
            .shouldMatch("src/main/java/aggregation/pipelines/join1to1/TutorialOutput.txt");
    }

    @Test
    @DisplayName("Test that multi-field join aggregation pipeline matches output")
    void TestJoinMultiField() {
        var example = new aggregation.pipelines.joinMultiField.Tutorial();
        example.loadSampleData();
        var output = example.runTutorial();

        Expect.that(output)
            .shouldMatch("src/main/java/aggregation/pipelines/joinMultiField/TutorialOutput.txt");
    }
}
